### PR TITLE
Feature/share publication

### PR DESCRIPTION
**Pull request by ismailIBNMAJAH**
# Feature: Share Publication

## Summary
This feature allows users to share a publication directly, uploading and sharing the file (photo or video) itself instead of just its URL.

---

### Backend Changes
- **Function `handleShare`:**  
  A new backend method `handleShare` has been implemented.  
  - This function handles the downloading of the media file, storing it in the appropriate directory, and initiating the sharing intent.
  - It supports both photos (`MediaType.PHOTO`) and videos (`MediaType.VIDEO`).

### UI Changes
- **Share Button:**  
  - Added a **Share Button** in the `VideoScreen` for each publication.  
  - The button triggers the `handleShare` function with the publication details.

---

## Testing
Comprehensive testing was performed for both the backend method and the UI:
1. **UI Tests:**
   - Verified the visibility of the Share Button for each publication.
   - Ensured proper interactions when clicking the Share Button.
2. **Backend Unit Tests:**
   - Tested `handleShare` for both successful and failure scenarios:
     - Successful sharing of photo and video.
     - Handling of failed downloads and empty response bodies.

---

## Screenshots
sharePub

sharePub2

### CODE REVIEW

File: app/src/androidTest/java/com/github/se/eduverse/ui/videos/VideoScreenTest.kt [hunk 1]
```diff
@@ -506,4 +513,92 @@ class VideoScreenTest {
...
+    composeTestRule.onNodeWithTag("ShareButton_0").assertExists().assertIsDisplayed()
+
+    composeTestRule.onNodeWithTag("ShareButton_0").performClick()
+
+    composeTestRule.waitForIdle()
+
+    verify { handleShare(publication = publication, context = any()) }
+
+    verify { Log.d("SHARE", "Share button clicked for publication: ${publication.id}") }
+  }
```
Comment (MathieuxC) [inline app/src/androidTest/java/com/github/se/eduverse/ui/videos/VideoScreenTest.kt:603]: tests seem efficent and complete

File: app/src/main/java/com/github/se/eduverse/ui/videos/Videos.kt [hunk 2]
```diff
@@ -227,6 +257,26 @@ fun VideoScreen(
...
+                                      Modifier.align(Alignment.CenterEnd)
+                                          .offset(y = 192.dp)
+                                          .padding(12.dp)
+                                          .testTag("ShareButton_$page")) {
+                                    Icon(
+                                        imageVector = Icons.Default.Share,
+                                        contentDescription = "Share",
+                                        tint = Color.White,
+                                        modifier = Modifier.size(48.dp))
+                                  }
```
Comment (MathieuxC) [inline app/src/main/java/com/github/se/eduverse/ui/videos/Videos.kt:279]: lgtm

File: app/src/main/java/com/github/se/eduverse/ui/videos/Videos.kt [hunk 3]
```diff
@@ -471,3 +522,76 @@ fun PhotoItem(
...
+    CoroutineScope(ioDispatcher).launch {
+      try {
+        val bytes = downloadBytes(publication.mediaUrl, client)
+        val mediaFile = createMediaFile(context, publication, bytes)
+        val uri: Uri =
+            FileProvider.getUriForFile(context, "${context.packageName}.fileprovider", mediaFile)
+        val shareIntent = createShareIntent(context, publication, uri)
+        withContext(mainDispatcher) {
+          context.startActivity(Intent.createChooser(shareIntent, null))
+          Toast.makeText(context, "Partage lancé", Toast.LENGTH_SHORT).show()
```
Comment (MathieuxC) [inline app/src/main/java/com/github/se/eduverse/ui/videos/Videos.kt:587]: the text should be in english

File: app/src/main/java/com/github/se/eduverse/ui/videos/Videos.kt [hunk 4]
```diff
@@ -471,3 +522,76 @@ fun PhotoItem(
...
+        }
+      } catch (e: Exception) {
+        e.printStackTrace()
+        withContext(mainDispatcher) {
+          Toast.makeText(context, "Erreur lors du partage: ${e.message}", Toast.LENGTH_SHORT).show()
```
Comment (MathieuxC) [inline app/src/main/java/com/github/se/eduverse/ui/videos/Videos.kt:592]: same here

**Review by MathieuxC — APPROVED** [review]
You create some toasts with french text so please replace it, otherwise lgtm

File: app/src/main/java/com/github/se/eduverse/ui/videos/Videos.kt (hunk 3 above)
Comment (ismailIBNMAJAH) [reply app/src/main/java/com/github/se/eduverse/ui/videos/Videos.kt]: well seen, it's good it's settled

**Pull request merged**